Checking for None instead of falsyness

abs007 opened the pull request:
Metadata API: Checking for None instead of falsyness for unrecognized_fields

Fixes #1937

Description of the changes being introduced by the pull request:

Initialization of unrecognized_fields acts surprisingly when the input
container is empty. Hence, We're checking for None instead of falsyness.

Signed-off-by: Abhisman Sarkar <abhisman.sarkar@gmail.com>

abs007 committed:
Metadata API: Checking for None instead of falsyness

Fixes #1937

Description of the changes being introduced by the pull request:

Initialization of unrecognized_fields acts surprisingly when the input
container is empty. Hence, We're checking for None instead of falsyness.

Signed-off-by: Abhisman Sarkar <abhisman.sarkar@gmail.com>

lukpueh reviewed (commented):
Thanks for the patch, @abs007! Two style comments:

1. There is no need to add PR template wording such as *"Description of the changes being introduced by the pull request:"* to the commit message
2. IMO the following construct is a bit easier on the eyes...

```python
if unrecognized_fields is None:
    unrecognized_fields = {}

self.unrecognized_fields = unrecognized_fields
```

abs007 commented:
Agreed. That does look better. I'll make the changes

abs007 committed:
Metadata API: Checking for None instead of falsyness for unrecognized_fields

Fixes #1937

Initialization of unrecognized_fields acts surprisingly when the input
container is empty. Hence, We're checking for None instead of falsyness.

Signed-off-by: Abhisman Sarkar abhisman.sarkar@gmail.com

abs007 committed:
Metadata API: Checking for None instead of falsyness for unrecognized_fields

Fixes #1937

Initialization of unrecognized_fields acts surprisingly when the input
container is empty. Hence, We're checking for None instead of falsyness.

Signed-off-by: Abhisman Sarkar <abhisman.sarkar@gmail.com>

abs007 commented:
I had a formatting issue earlier but I resolved that. For some reason, the signing error persists even though I've added my signature. 
Could you look into this, thanks.

abs007 commented:
Closing this pr and opening a new one for this issue as I think that might solve the signature issue.

lukpueh commented:
Hi @abs007, usually there is no need to close/re-open pull requests, as long as you don't need to switch the branch of the PR. I recommend reading our [git history guidelines](https://github.com/secure-systems-lab/lab-guidelines/blob/master/git-history.md). It has some information about how to rewrite the history in a given git branch.

abs007 commented:
Thanks for the info. 
I unfortunately couldn't understand a part in the example provided by @awwad for rebasing commits, made on Dec 6, 2018. 
Specifically, the third paragraph "I've fixed this using `git rebase -i` ..."  

![Inkedgit_LI](https://user-images.githubusercontent.com/53483248/165890953-a9000091-a9ef-4f44-8be7-80eccc9d0a39.jpg)

Could you provide resources to understand this, thanks.

lukpueh commented:
> Could you provide resources to understand this, thanks.

Did you check out the resources about `git rebase` in the git history guide I shared with you above?

lukpueh commented:
Don't worry too much about @awwad's comment, it is not supposed to be a general tutorial for `git rebase`.

abs007 commented:
> Did you check out the resources about git rebase in the git history guide I shared with you above?

Haven't gone through that. Will do.
